R/plotPValHist.R

```diff
@@ -31,104 +30,146 @@
+            cx.data   <- subset(P.Val, select = pval)
+            var.annot <- subset(P.Val, select = -c(pval))
+            plotlist  <- canvasXpress::canvasXpress(data                = cx.data,
+                                                   varAnnot             = var.annot,
+                                                   histogramData        = TRUE,
+                                                   graphType            = "Scatter2D",
+                                                   colors               = color,
+                                                   transparency         = transparency,
+                                                   title                = title,
+                                                   xAxisTitle           = "P-value",
+                                                   yAxisTitle           = "Count",
+                                                   hideHistogram        = FALSE,
+                                                   showHistogramDensity = FALSE,
+                                                   showLegend           = FALSE,
+                                                   segregateVariablesBy = list("levels"),
+                                                   events               = events)
         } else {
-            for (i in 1:NumSamples) {
-                s <- SampNames[i]
-                MyPVal <- dplyr::filter(P.Val, grepl(s, Levels))
-                cx.data <- subset(MyPVal, select = Pval)
-                var.annot <- subset(MyPVal, select = -c(Pval))
-                Hist_Pval <- canvasXpress::canvasXpress(data                    = cx.data,
-                                                        varAnnot                = var.annot,
-                                                        histogramData           = TRUE,
-                                                        graphType               = "Scatter2D",
-                                                        colors                  = color,
-                                                        title                   = paste(title, "\n", s),
-                                                        xAxisTitle              = "P-value",
-                                                        yAxisTitle              = "Count",
-                                                        hideHistogram           = FALSE,
-                                                        showHistogramDensity    = FALSE,
-                                                        showLegend              = FALSE,
-                                                        segregateVariablesBy    = list("Levels"))
-
-                plotlist[[i]] = Hist_Pval
+            for (i in 1:samples_num) {
+                sample <- sample_names[i]
+                pval_subset <- dplyr::filter(P.Val, grepl(sample, levels))
+                cx.data <- subset(pval_subset, select = pval)
```

Comment:
use **dplyr select** method

Reply:
As discussed with Connie, We need to try and use external functions only when required. Will retain this for now. If it has to absolutely be used anywhere, then we can refactor these functions later on to use dplyr equivalent function